### PR TITLE
Fix/data changes

### LINKED ISSUE #4391: User List - CV links go into the void
grafik

For some reason, alltho I checked I'm not looking for work, my CV would be displayed, and since I have none provided, it goes into the void (c.gitcoin.co)

## Fix:
Check for content, do not display href if not available.

### PR DESCRIPTION
**Pull request by octavioamu**
##### Description

- remove sensitive data
- api only return needed data
- fix resume / job status condition #4391
- don't show hidden profiles

##### Refers/Fixes

fix #4391
##### Testing

Tested the diff user data configurations

### CODE REVIEW

File: app/dashboard/views.py [hunk 1]
```diff
@@ -820,33 +822,42 @@ def users_fetch(request):
     all_users = []
     for user in all_pages.page(page):
         profile_json = {}
-        profile_json = user.to_standard_dict()
-        if user.avatar_baseavatar_related.exists():
-            user_avatar = user.avatar_baseavatar_related.first()
-            profile_json['avatar_id'] = user_avatar.pk
-            profile_json['avatar_url'] = user_avatar.avatar_url
+        # profile_json = user.to_standard_dict()
```
Comment (danlipert) [inline app/dashboard/views.py]: Extra comment

File: app/dashboard/views.py [hunk 2]
```diff
@@ -776,7 +776,9 @@ def users_fetch(request):
     else:
         network = 'rinkeby'
 
-    user_list = Profile.objects.prefetch_related('fulfilled', 'leaderboard_ranks', 'feedbacks_got').order_by(order_by)
+    user_list = Profile.objects.prefetch_related(
```
Comment (danlipert) [inline app/dashboard/views.py]: Can we call this `profile_list` instead since there is already a `User` model?

File: app/dashboard/views.py [hunk 3]
```diff
@@ -820,33 +822,42 @@ def users_fetch(request):
...
-        previously_worked_with = 0
-        if profile:
-            previously_worked_with = BountyFulfillment.objects.filter(
-                bounty__bounty_owner_github_username__iexact=profile.handle,
-                fulfiller_github_username__iexact=user.handle,
-                bounty__network=network,
-                bounty__accepted=True
-            ).count()
-
+        profile_json['id'] = user.id
```
Comment (danlipert) [inline app/dashboard/views.py]: `profile_json = {k: user['k'] for k in ['id', 'actions_count', 'created_on'... etc]}` and then `profile_json['job_status'] = user.job_status_verbose if user.job_search_status else None` as the only unique one. Less lines is (almost) always good :)

File: app/dashboard/views.py [hunk 4]
```diff
@@ -820,33 +822,42 @@ def users_fetch(request):
...
         profile_json['avg_rating'] = user.get_average_star_rating
         # profile_json['bounties'] = user.get_quarterly_stats
         profile_json['is_org'] = user.is_org
+        if user.avatar_baseavatar_related.exists():
+            user_avatar = user.avatar_baseavatar_related.first()
+            profile_json['avatar_id'] = user_avatar.pk
+            profile_json['avatar_url'] = user_avatar.avatar_url
+        if user.data:
+            user_data = user.data
+            print(user_data)
```
Comment (danlipert) [inline app/dashboard/views.py]: extra `print()`

File: app/dashboard/views.py (hunk 2 above)
Comment (octavioamu) [reply app/dashboard/views.py]: true maybe in plural? `profiles_list` ?

Comment (danlipert) [reply app/dashboard/views.py]: I like `profile_list` personally but either one is ok 👍

File: app/dashboard/views.py [hunk 5]
```diff
@@ -802,51 +804,45 @@ def users_fetch(request):
...
-        previously_worked_with = 0
-        if profile:
-            previously_worked_with = BountyFulfillment.objects.filter(
-                bounty__bounty_owner_github_username__iexact=profile.handle,
-                fulfiller_github_username__iexact=user.handle,
-                bounty__network=network,
-                bounty__accepted=True
-            ).count()
-
+        profile_json = {k: getattr(user, k) for k in
```
Comment (danlipert) [inline app/dashboard/views.py]: slick! Style-wise, lets try to align with PEP8 style - https://www.python.org/dev/peps/pep-0008/
```
profile_json = {
    k: getattr(user, k) for k in
    ['id', 'actions_count', 'created_on', 'handle', 'hide_profile',
     'show_job_status', 'job_location', 'job_salary', 'job_search_status',
     'job_type', 'linkedin_url', 'resume', 'remote', 'keywords',
     'organizations', 'is_org']}
```

You can check with http://pep8online.com/ - I want to integrate a python stylechecker into our makefile/tests as well

**Comment by octavioamu** [comment]
Ready to review @danlipert

**Pull request merged**